src/lib/hooks/useFocus/index.svelte.ts

```diff
@@ -0,0 +1,178 @@
+import { getWindow, isElement, isHTMLElement } from '@floating-ui/utils/dom';
+import type { FloatingContext } from '../useFloating/index.svelte.js';
+import { getDocument } from '$lib/utils/get-document.js';
+import { activeElement } from '$lib/utils/active-element.js';
+import type { OpenChangeReason } from '$lib/types.js';
+import { isVirtualPointerEvent } from '$lib/utils/is-virtual-pointer-event.js';
+import { getTarget } from '$lib/utils/get-target.js';
+import { isSafari } from '$lib/utils/is-safari.js';
```

Comment:
Maybe consider grouping the `is{Platform}` utilities into a single util script file called `is-platform.ts`?

This includes: android, safari, mac, etc

Reply:
Yeah, that's a great suggestion, will do that. It's indeed getting a little crazy.

Reply:
Done